src/foam/dao/FileJournal.js

```diff
@@ -171,14 +152,27 @@ foam.CLASS({
           throw new RuntimeException(t);
         }
       `
+    },
+    {
+      class: 'Object',
+      name: 'lock',
+      javaType: 'java.util.concurrent.locks.Lock',
+      documentation: 'The lock to synchronize writing records on.  If multiple file journals write to the same file they should share a lock.',
+      javaFactory: 'return new java.util.concurrent.locks.ReentrantLock();',
```

Comment:
Could you please make this a fair lock to ensure that the ordering is correct when we read back the records? This will also be useful to guarantee that if you stop reading at any point in the journal, you have the correct and complete state of the system up until that point.

Reply:
Please disregard my previous comment :)  From our discussion this afternoon: this wouldn't matter because it would never occur (famous last words) that two records being written in the same journal for two different DAOs are dependent on the order of being written and read.

The reason that this wouldn't occur is because the only that would need to happen is if the `put_` is being overridden in a decorator and the logic `put`s on two DAOs. `put` is `synchronized` and there only ever exists one object for the DAO in the context.

This has to be noted somewhere in the code.

Cheers!